### ISSUE TITLE
Tests using MOI 0.9.21 are broken

### ISSUE THREAD

**Issue by guilhermebodin** [jump-dev/Dualization.jl#108]
https://github.com/jump-dev/Dualization.jl/runs/2632781531?check_suite_focus=true

When testing Dualization with MOI 0.9.21 things got ugly.

**Comment by odow** [comment]
You can't hard-code the constraint indices like this. They need to be mapped through the appropriate layers.
https://github.com/jump-dev/Dualization.jl/blob/1a51249bbe4741429e822d6ef37445160f7315e6/test/Tests/test_dualize_conic_linear.jl#L37

**Comment by guilhermebodin** [comment]
Indeed! I am going to investigate how to migrate these weak hard-coded logics to a more robust ones
